bool safety
int32 danger_index
int32 section_index